whylog_vim/gui.py

```diff
@@ -0,0 +1,209 @@
+from whylog import LineSource, FrontInput
+
+from mock import MagicMock, patch
+import os.path
+import vim
+from whylog_vim.input_reader.teacher_reader import get_button_name, filter_comments
+from whylog_vim.consts import WindowTypes, WindowSizes
+
+
+
+class WindowContext(object):
+
+    def __init__(self, gui):
+        self.gui = gui
+
+    def __enter__(self):
+        vim.command('setlocal modifiable')
+        return vim.current.buffer
+
+    def __exit__(self, type_, value, traceback):
+        vim.command('setlocal nomodifiable')
+
+
+class VimEditor():
```

Comment:
Divide this class into smaller. I see a few responsibilities that can indicate division: 
- windows creating (create_case_window, create_input_window ...)
- windows management (split_window, resize ...)
- cursor (cursor_at_output, cursor_at_teacher...)
- movements (go_to_output_window, go_to_line, go_to_offset, get_col...)
- extracting line data (get_current_filename, get_line_number, get_offset...)